File: leetcode/hashtable/dart/contains_duplicate.dart
class Solution {
  bool containsDuplicate(List<int> nums) {
    final hashmap = <int, bool>{};
    for (var i = 0; i < nums.length; i++) {
      if (hashmap[nums[i]] == true) {
        return true;
      }
      hashmap[nums[i]] = true;
    }
    return false;
  }
}

class Solution2 {
  bool containsDuplicate(List<int> nums) {
    return nums.toSet().length != nums.length;
  }
}

void main(List<String> args) {
  final object = Solution();
  final nums = [1, 2, 3, 3, 4];
  print(object.containsDuplicate(nums));
}

File: leetcode/hashtable/go/contains_duplicate.go
package hashmap

func containsDuplicate(nums []int) bool {
	hashMap := make(map[int][]int)

	for index, value := range nums {
		hashMap[value] = append(hashMap[value], index)

	}
	for index, value := range hashMap {
		hashMap[index] = value
	}
	return len(hashMap) == len(nums)
}

func containsDuplicate2(nums []int) bool {
	hashmap := make(map[int]bool)
	for _, i := range nums {
		if hashmap[nums[i]] == true {
			return true
		}
		hashmap[nums[i]] = true
	}
	return false
}
